### PR TITLE
Update config for snap

### PR DESCRIPTION
**Pull request by RikkiGibson**

### CODE REVIEW

**Review by dibarbet — APPROVED** [review]

File: eng/config/PublishData.json [hunk 1]
```diff
@@ -214,11 +214,19 @@
       "vsBranch": "rel/d16.9",
       "vsMajorVersion": 16
     },
-    "master-vs-deps": {
+    "release/dev16.10-vs-deps": {
       "nugetKind": ["Shipping", "NonShipping"],
       "version": "3.10.*",
       "packageFeeds": "default",
       "channels": [ "dev16.10", "dev16.10p1" ],
```
Comment (JoeRobich) [inline eng/config/PublishData.json:221]:
```suggestion
      "channels": [ "dev16.10p1" ],
```

**Review by JoeRobich — APPROVED** [review]

File: eng/config/PublishData.json (hunk 1 above)
Comment (RikkiGibson) [reply eng/config/PublishData.json:221]: what does this change do?

Comment (JoeRobich) [reply eng/config/PublishData.json:221]: Perhaps I misremember but I believe we let the most forward branch publish to the `dev16.x` channel and let preview branches continue to publish to the `dev16.xpy` channel.

Comment (RikkiGibson) [reply eng/config/PublishData.json]: works for me!

Comment (JoeRobich) [reply eng/config/PublishData.json]: That being said, from looking at what the channels are updating (https://github.com/dotnet/versions/tree/master/build-info/dotnet/roslyn/), I am not sure we need to be maintaining channels anyway. I believe maestro and darc have surplanted this bit of infra.

Comment (jaredpar) [reply eng/config/PublishData.json]: Yeah I'm not sure we need this anymore. Think thta was mostly useful for the manual nuget upload which we no longer do.

**Pull request merged**